docs/benchmarking-workflow-corefx.md

```diff
@@ -0,0 +1,253 @@
+
+## Local CoreCLR Build
+
+Sometimes you might be willing to run the benchmarks using not only local CoreFX but also local CoreCLR build.
+
+Thanks to the simplicity of CoreRun, all you need to do is to copy all the runtimes files into the folder with CoreRun.
+
+But before you do it, you should copy/remeber the version of CoreCLR reported by BenchmarkDotNet:
+
+```log
+BenchmarkDotNet=v0.11.3.1003-nightly, OS=Windows 10.0.17763.253 (1809/October2018Update/Redstone5)
+Intel Xeon CPU E5-1650 v4 3.60GHz, 1 CPU, 12 logical and 6 physical cores
+.NET Core SDK=3.0.100-preview-009812
+  [Host]     : .NET Core 3.0.0-preview-27122-01 (CoreCLR 4.6.27121.03, CoreFX 4.7.18.57103), 64bit RyuJIT
+  Job-SBBRNM : .NET Core ? (CoreCLR 4.6.27416.73, CoreFX 4.7.19.11901), 64bit RyuJIT
+```
+
+As you can see, before the "update" the version for the Job (not the Host process) was `4.6.27416.73`.
+
+The CoreFX build system has a built-in support for that, exposed by the [CoreCLROverridePath](https://github.com/dotnet/corefx/blob/0e7236fda21a07302b14030c82f79bb981c723a6/Documentation/project-docs/developer-guide.md#testing-with-private-coreclr-bits) build parameter:
+
+```cmd
+C:\Projects\corefx> build -Release /p:CoreCLROverridePath="C:\Projects\coreclr\bin\Product\Windows_NT.x64.Release"
+```
+
+After the update, the reported CoreCLR version should be changed:
+
+```log
+  [Host]     : .NET Core 3.0.0-preview-27122-01 (CoreCLR 4.6.27121.03, CoreFX 4.7.18.57103), 64bit RyuJIT
+  Job-CYVJFZ : .NET Core ? (CoreCLR 4.6.27517.0, CoreFX 4.7.19.11901), 64bit RyuJIT
+```
+
+**It's very important to validate the reported version numbers** to make sure that you are running the benchmarks using the versions you think that you are using.
+
+## Benchmarking new API
+
+When developing new CoreFX features, we should be thinking about the performance from day one. One part of doing this is writing benchmarks at the same time when we write our first unit tests. Keeping the benchmarks in a separate repository makes it a little bit harder to run the benchmarks against new CoreFX API, but it's still very easy.
+
+### Reference
+
+When you develop a new feature, whether it's a new method/type/library in CoreFX all you need to do is to build it in Release and just reference the produced `.dll` from the [MicroBenchmarks.csproj](../src/benchmarks/micro/MicroBenchmarks.csproj) project file.
+
+The easiest way to do it is to open [MicroBenchmarks.sln](../src/benchmarks/micro/MicroBenchmarks.sln) with Visual Studio, right click on the [MicroBenchmarks](../src/benchmarks/micro/MicroBenchmarks.csproj) project file, select "Add", then "Reference..." and in the new Dialog Window click "Browse" in the left bottom corner. From the File Picker, choose the new library and click "Add". Please don't forget to Save the changes (Ctrl+Shift+S). From this moment you should be able to consume new public types and methods exposed by the referenced library.
```

Comment:
`choose the new library and click "Add"`

Do we add a reference to the implementation assembly or the reference assembly of the new library?

Reply:
good question! I added "implementation" word to the description.

Was:

> reference the produced `.dll`

Is:

> reference the produced implementation `.dll`